api.py

```diff
@@ -65,10 +70,18 @@ def init_rollbar():
         # send exceptions from `app` to rollbar, using flask's signal system.
         got_request_exception.connect(rollbar.contrib.flask.report_exception, app)
 
+
 @app.before_request
 def validate_api_key():
-    if not request.args.get('apikey', None) == app.config['APIKEY']:
-        abort(403)
+    apikey = request.args.get('apikey', request.headers.get('X-Api-Key', None)) # get the API key from a request param or a header
+
+    sql = get_table('Sesh_RMC_Account').select().where(sqlalchemy.text('API_KEY = :k')).limit(1)
+    result = app.engine.execute(sql, k=apikey)
```

Comment:
not sure if I understand what could be cached. we need to authenticate every request so no caching. 
It is one super simple SQL query. No need for another dependency and database... 
You could check if there is an index on the API_KEY field. - though I assume there will not be many requests (how many sites do you expect)